src/utils.ts

```diff
@@ -63,11 +63,14 @@ export const collectCopyTargets = async (
 
       // https://github.com/vladshcherbin/rollup-plugin-copy/blob/507bf5e99aa2c6d0d858821e627cb7617a1d9a6d/src/index.js#L32-L35
       const { base, dir } = path.parse(matchedPath)
+
+      const dirClean = dir.replace(/^(?:\.\.\/)+/, '')
+      const destClean = dirClean.replace(dirClean.split('/')[0], dest).replace(/^\/+|\/+$/g, '')
+
       const destDir =
         flatten || (!flatten && !dir)
           ? dest
-          : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            dir.replace(dir.split('/')[0]!, dest)
+          : destClean
```

Comment:
I guess this is same with the following code.
```js
path.resolve(dest, path.normalize(path.join(dir, '../')))
```

Also would you add some test cases here?
https://github.com/sapphi-red/vite-plugin-static-copy/blob/08a2a9b7559c32f1c12490bc9361485406b3c534/test/testcases.ts#L9

Reply:
> ```javascript
> path.resolve(dest, path.normalize(path.join(dir, '../')))
> ```
The result with this code is different from the one in the PR.
The goal is to target the `build.outDir` defined in vite.config. Your suggestion ended up copying the files somewhere else entirely. Is the code suggested in the PR not good? It does the job.

> Also would you add some test cases here?

I'm sorry, I don't know how to do that.

Reply:
> The result with this code is different from the one in the PR.

Ah, my bad I was misunderstanding. Please forget that.

> I'm sorry, I don't know how to do that.

I added the tests. It seems your change breaks some behaviors.